ballerina/context.bal

```diff
@@ -138,25 +138,27 @@ public isolated class Context {
     }
 
     isolated function getNextInterceptor(Field 'field) returns (readonly & Interceptor)? {
-        lock {
-            Engine? engine = self.getEngine();
-            if engine is Engine {
-                (readonly & Interceptor)[] interceptors = engine.getInterceptors();
-                if interceptors.length() > self.nextInterceptor {
-                    (readonly & Interceptor) next = interceptors[self.nextInterceptor];
-                    self.nextInterceptor += 1;
-                    return next;
-                }
-                int nextFieldInterceptor = self.nextInterceptor - engine.getInterceptors().length();
-                if 'field.getFieldInterceptors().length() > nextFieldInterceptor {
-                    readonly & Interceptor next = 'field.getFieldInterceptors()[nextFieldInterceptor];
-                    self.nextInterceptor += 1;
-                    return next;
+        Engine? engine = self.getEngine();
+        if engine is Engine {
+            (readonly & Interceptor)[] interceptors = engine.getInterceptors();
+            if interceptors.length() > self.getInterceptorCount() {
+                (readonly & Interceptor) next = interceptors[self.getInterceptorCount()];
+                if !isGlobalInterceptor(next) && 'field.getPath().length() > 1 {
```

Comment:
Is this logic accurate? We need a better way to figure out how to omit the non-global interceptors without considering the path. 

We might need to apply this to other service types as well. This logic won't work in those scenarios.

I think we might have to keep track of the interceptors and their respective types somewhere else. If this is too complex to handle, we can go ahead with this logic for now and improve later on.

Reply:
+1